tests/unit_tests/PluginTelemetryTest.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+
+namespace Payplug;
+use Payplug;
+use \Payplug\Core\HttpClient;
+use Payplug\Core\APIRoutes;
+use PHPUnit\Framework\TestCase;
+
+class PluginTelemetryTest extends TestCase
+{
+
+    private $_requestMock;
+    private $_httpClient;
+
+    /**
+     * @before
+     */
+    protected function setUpTest()
+    {
+        $this->_configuration = new Payplug\Payplug('abc');
+        $this->_httpClient = new HttpClient(new \Payplug\Payplug('abc'));
+        Payplug\Payplug::setDefaultConfiguration($this->_configuration);
+
+        $this->_requestMock = $this->createMock('Payplug\Core\IHttpRequest');
+        Payplug\Core\HttpClient::$REQUEST_HANDLER = $this->_requestMock;
+    }
+
+    /**
+     * Verify the behavior of send() method in the PluginTelemetry class
+     * with a mocked API URL that raises an exception
+     *
+     * @throws Exception\ConnectionException
+     * @throws Exception\HttpException
+     * @throws Exception\UnexpectedAPIResponseException
+     */
+    public function testSendWithBadMockedURL()
```

Comment:
You have 4 use cases whose raise Exception in the send method, I think you should have at least 4 tests exceptions.

Reply:
Thank you for bringing up the issue of handling exceptions in our code and the corresponding test cases.
Regarding your suggestion,  we completely agree. You raised a valid concern. 
In terms of testing our code, i'd like to provide some additional context. As of now, due to the absence of the deployed API URL, we are limited in our ability to fully test these exception scenarios in a real-world environment.
Furthermore, we don't want to hardcode URL, especially those from our intern gitlab. 

We plan to use configuration file (cms sides) so that we can specify the API URL as needed, making it easier to switch between different environments such as developing,  testing and production. This approch allows us to adjust the URL when the API is deployed.

However, we recognize the importance of starting the testing process early and maintaining a structured approach. To address this, our immediate focus is on validating whether the URL provided is valid. This initial test case allows us to at least ensure that the URL is working or not.

We're actively thinking about  strategies to extend our test coverage as soon as the API is deployed and the URL becomes available. This includes building upon the foundation of the URL validation test case and progressively integrating additional test cases as we gain access to the actual API endpoints.

Reply:
OK so I'll check that later!